From 39c41e1221ec2d7f3c517bb5ea08db4b57eec43a Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
Date: Thu, 26 Sep 2019 10:43:15 +0200
Subject: [PATCH] edma: dts: Channel 20 and 21 are reserved for pure memory
 transfers

and should not be used for device access.
"ti,edma-memcpy-channels = <20 21>;"

Furthermore use correct pin configuration (EVENT_INTR1 & EVENT_INTR2)
for dma channels.

Signed-off-by: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
---
 arch/arm/boot/dts/am335x-pac-768_330x.dtsi | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pac-768_330x.dtsi b/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
index fb56178..603ab29 100644
--- a/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
+++ b/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
@@ -51,13 +51,13 @@
 		/*
 		 * Crossbar Mapped Channels
 		 *
-		 * 28: pi_x_dma_event_intr0
 		 * 29: pi_x_dma_event_intr1
+		 * 30: pi_x_dma_event_intr2
 		 *
-		 * Map 28 and 29 to Open Channels 20 and 21
+		 * Map 30 and 29 to open channels 12 and 13
 		 *
 		 */
-		dmas = <&edma_xbar 20 0 28>, <&edma_xbar 21 0 29>;
+		dmas = <&edma_xbar 12 0 30>, <&edma_xbar 13 0 29>;
 		dma-names = "rx", "tx";
 
 		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
@@ -239,8 +239,8 @@
 		pinctrl-single,pins = <
 			AM33XX_LCD_DATA6_GPIO2_12(PIN_INPUT)
 			AM33XX_LCD_AC_BIAS_EN_GPIO2_25(PIN_INPUT)
-			AM33XX_XDMA_EVENT_INTR0(PIN_INPUT)
 			AM33XX_XDMA_EVENT_INTR1(PIN_INPUT)
+			AM33XX_SPI0_CS1_XDMA_EVENT_INTR2(PIN_INPUT)
 		>;
 	};
 };
-- 
2.7.4

